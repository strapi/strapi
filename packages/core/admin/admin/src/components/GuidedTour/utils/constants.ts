const GUIDED_TOUR_REQUIRED_ACTIONS = {
  contentTypeBuilder: {
    createSchema: 'didCreateContentTypeSchema',
    addField: 'didAddFieldToSchema',
  },
  contentManager: {
    createContent: 'didCreateContent',
  },
  apiTokens: {
    createToken: 'didCreateApiToken',
    copyToken: 'didCopyApiToken',
  },
  strapiCloud: {},
} as const;

export { GUIDED_TOUR_REQUIRED_ACTIONS };
