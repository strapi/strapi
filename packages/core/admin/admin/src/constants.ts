import { PermissionMap } from './types/permissions';

import type { StrapiAppSettingLink } from './core/apis/router';

export const ADMIN_PERMISSIONS_CE = {
  contentManager: {
    main: [],
    collectionTypesConfigurations: [
      {
        action: 'plugin::content-manager.collection-types.configure-view',
        subject: null,
      },
    ],
    componentsConfigurations: [
      {
        action: 'plugin::content-manager.components.configure-layout',
        subject: null,
      },
    ],
    singleTypesConfigurations: [
      {
        action: 'plugin::content-manager.single-types.configure-view',
        subject: null,
      },
    ],
  },
  marketplace: {
    main: [{ action: 'admin::marketplace.read', subject: null }],
    read: [{ action: 'admin::marketplace.read', subject: null }],
  },
  settings: {
    roles: {
      main: [
        { action: 'admin::roles.create', subject: null },
        { action: 'admin::roles.update', subject: null },
        { action: 'admin::roles.read', subject: null },
        { action: 'admin::roles.delete', subject: null },
      ],
      create: [{ action: 'admin::roles.create', subject: null }],
      delete: [{ action: 'admin::roles.delete', subject: null }],
      read: [{ action: 'admin::roles.read', subject: null }],
      update: [{ action: 'admin::roles.update', subject: null }],
    },
    users: {
      main: [
        { action: 'admin::users.create', subject: null },
        { action: 'admin::users.read', subject: null },
        { action: 'admin::users.update', subject: null },
        { action: 'admin::users.delete', subject: null },
      ],
      create: [{ action: 'admin::users.create', subject: null }],
      delete: [{ action: 'admin::users.delete', subject: null }],
      read: [{ action: 'admin::users.read', subject: null }],
      update: [{ action: 'admin::users.update', subject: null }],
    },
    webhooks: {
      main: [
        { action: 'admin::webhooks.create', subject: null },
        { action: 'admin::webhooks.read', subject: null },
        { action: 'admin::webhooks.update', subject: null },
        { action: 'admin::webhooks.delete', subject: null },
      ],
      create: [{ action: 'admin::webhooks.create', subject: null }],
      delete: [{ action: 'admin::webhooks.delete', subject: null }],
      read: [
        { action: 'admin::webhooks.read', subject: null },
        // NOTE: We need to check with the API
        { action: 'admin::webhooks.update', subject: null },
        { action: 'admin::webhooks.delete', subject: null },
      ],
      update: [{ action: 'admin::webhooks.update', subject: null }],
    },
    'api-tokens': {
      main: [{ action: 'admin::api-tokens.access', subject: null }],
      create: [{ action: 'admin::api-tokens.create', subject: null }],
      delete: [{ action: 'admin::api-tokens.delete', subject: null }],
      read: [{ action: 'admin::api-tokens.read', subject: null }],
      update: [{ action: 'admin::api-tokens.update', subject: null }],
      regenerate: [{ action: 'admin::api-tokens.regenerate', subject: null }],
    },
    'transfer-tokens': {
      main: [{ action: 'admin::transfer.tokens.access', subject: null }],
      create: [{ action: 'admin::transfer.tokens.create', subject: null }],
      delete: [{ action: 'admin::transfer.tokens.delete', subject: null }],
      read: [{ action: 'admin::transfer.tokens.read', subject: null }],
      update: [{ action: 'admin::transfer.tokens.update', subject: null }],
      regenerate: [{ action: 'admin::transfer.tokens.regenerate', subject: null }],
    },
    'project-settings': {
      read: [{ action: 'admin::project-settings.read', subject: null }],
      update: [{ action: 'admin::project-settings.update', subject: null }],
    },
    plugins: {
      main: [{ action: 'admin::marketplace.read', subject: null }],
      read: [{ action: 'admin::marketplace.read', subject: null }],
    },
  },
} satisfies Partial<PermissionMap>;

export const HOOKS = {
  /**
   * Hook that allows to mutate the displayed headers of the list view table
   * @constant
   * @type {string}
   */
  INJECT_COLUMN_IN_TABLE: 'Admin/CM/pages/ListView/inject-column-in-table',

  /**
   * Hook that allows to mutate the CM's collection types links pre-set filters
   * @constant
   * @type {string}
   */
  MUTATE_COLLECTION_TYPES_LINKS: 'Admin/CM/pages/App/mutate-collection-types-links',

  /**
   * Hook that allows to mutate the CM's edit view layout
   * @constant
   * @type {string}
   */
  MUTATE_EDIT_VIEW_LAYOUT: 'Admin/CM/pages/EditView/mutate-edit-view-layout',

  /**
   * Hook that allows to mutate the CM's single types links pre-set filters
   * @constant
   * @type {string}
   */
  MUTATE_SINGLE_TYPES_LINKS: 'Admin/CM/pages/App/mutate-single-types-links',
};

export interface SettingsMenuLink
  extends Omit<StrapiAppSettingLink, 'Component' | 'permissions' | 'licenseOnly'> {
  licenseOnly?: boolean;
}

export type SettingsMenu = {
  admin: SettingsMenuLink[];
  global: SettingsMenuLink[];
};

export const SETTINGS_LINKS_CE = (): SettingsMenu => ({
  global: [
    {
      intlLabel: { id: 'Settings.application.title', defaultMessage: 'Overview' },
      to: '/settings/application-infos',
      id: '000-application-infos',
    },
    {
      intlLabel: { id: 'Settings.webhooks.title', defaultMessage: 'Webhooks' },
      to: '/settings/webhooks',
      id: 'webhooks',
    },
    {
      intlLabel: { id: 'Settings.apiTokens.title', defaultMessage: 'API Tokens' },
      to: '/settings/api-tokens?sort=name:ASC',
      id: 'api-tokens',
    },
    {
      intlLabel: { id: 'Settings.transferTokens.title', defaultMessage: 'Transfer Tokens' },
      to: '/settings/transfer-tokens?sort=name:ASC',
      id: 'transfer-tokens',
    },
    {
      intlLabel: {
        id: 'global.plugins',
        defaultMessage: 'Plugins',
      },
      to: '/settings/list-plugins',
      id: 'plugins',
    },
    // If the Enterprise/Cloud feature is not enabled and if the config doesn't disable it, we promote the Enterprise/Cloud feature by displaying them in the settings menu.
    // Disable this by adding "promoteEE: false" to your `./config/admin.js` file
    ...(!window.strapi.features.isEnabled(window.strapi.features.SSO) &&
    window.strapi?.flags?.promoteEE
      ? [
          {
            intlLabel: { id: 'Settings.sso.title', defaultMessage: 'Single Sign-On' },
            to: '/settings/purchase-single-sign-on',
            id: 'sso-purchase-page',
            licenseOnly: true,
          },
        ]
      : []),
    ...(!window.strapi.features.isEnabled('cms-content-history') && window.strapi?.flags?.promoteEE
      ? [
          {
            intlLabel: { id: 'Settings.content-history.title', defaultMessage: 'Content History' },
            to: '/settings/purchase-content-history',
            id: 'content-history-purchase-page',
            licenseOnly: true,
          },
        ]
      : []),
  ],

  admin: [
    {
      intlLabel: { id: 'global.roles', defaultMessage: 'Roles' },
      to: '/settings/roles',
      id: 'roles',
    },
    {
      intlLabel: { id: 'global.users', defaultMessage: 'Users' },
      // Init the search params directly
      to: '/settings/users?pageSize=10&page=1&sort=firstname',
      id: 'users',
    },
    ...(!window.strapi.features.isEnabled(window.strapi.features.AUDIT_LOGS) &&
    window.strapi?.flags?.promoteEE
      ? [
          {
            intlLabel: { id: 'global.auditLogs', defaultMessage: 'Audit Logs' },
            to: '/settings/purchase-audit-logs',
            id: 'auditLogs-purchase-page',
            licenseOnly: true,
          },
        ]
      : []),
  ],
});
