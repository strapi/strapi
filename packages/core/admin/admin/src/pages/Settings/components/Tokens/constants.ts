export const API_TOKEN_TYPE = 'api-token';
export const TRANSFER_TOKEN_TYPE = 'transfer-token';
