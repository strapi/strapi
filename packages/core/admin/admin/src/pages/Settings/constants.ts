import type { RouteObject } from 'react-router-dom';

export const ROUTES_CE: RouteObject[] = [
  {
    lazy: async () => {
      const { ProtectedListPage } = await import('./pages/Roles/ListPage');

      return {
        Component: ProtectedListPage,
      };
    },
    path: 'roles',
  },
  {
    lazy: async () => {
      const { ProtectedCreatePage } = await import('./pages/Roles/CreatePage');

      return {
        Component: ProtectedCreatePage,
      };
    },
    path: 'roles/duplicate/:id',
  },
  {
    lazy: async () => {
      const { ProtectedCreatePage } = await import('./pages/Roles/CreatePage');

      return {
        Component: ProtectedCreatePage,
      };
    },
    path: 'roles/new',
  },
  {
    lazy: async () => {
      const { ProtectedEditPage } = await import('./pages/Roles/EditPage');

      return {
        Component: ProtectedEditPage,
      };
    },
    path: 'roles/:id',
  },
  {
    lazy: async () => {
      const { ProtectedListPage } = await import('./pages/Users/ListPage');

      return {
        Component: ProtectedListPage,
      };
    },
    path: 'users',
  },
  {
    lazy: async () => {
      const { ProtectedEditPage } = await import('./pages/Users/EditPage');

      return {
        Component: ProtectedEditPage,
      };
    },
    path: 'users/:id',
  },
  {
    lazy: async () => {
      const { ProtectedCreatePage } = await import('./pages/Webhooks/CreatePage');

      return {
        Component: ProtectedCreatePage,
      };
    },
    path: 'webhooks/create',
  },
  {
    lazy: async () => {
      const { ProtectedEditPage } = await import('./pages/Webhooks/EditPage');

      return {
        Component: ProtectedEditPage,
      };
    },
    path: 'webhooks/:id',
  },
  {
    lazy: async () => {
      const { ProtectedListPage } = await import('./pages/Webhooks/ListPage');

      return {
        Component: ProtectedListPage,
      };
    },
    path: 'webhooks',
  },
  {
    lazy: async () => {
      const { ProtectedListView } = await import('./pages/ApiTokens/ListView');

      return {
        Component: ProtectedListView,
      };
    },
    path: 'api-tokens',
  },
  {
    lazy: async () => {
      const { ProtectedCreateView } = await import('./pages/ApiTokens/CreateView');

      return {
        Component: ProtectedCreateView,
      };
    },
    path: 'api-tokens/create',
  },
  {
    lazy: async () => {
      const { ProtectedEditView } = await import('./pages/ApiTokens/EditView/EditViewPage');

      return {
        Component: ProtectedEditView,
      };
    },
    path: 'api-tokens/:id',
  },
  {
    lazy: async () => {
      const { ProtectedCreateView } = await import('./pages/TransferTokens/CreateView');

      return {
        Component: ProtectedCreateView,
      };
    },
    path: 'transfer-tokens/create',
  },
  {
    lazy: async () => {
      const { ProtectedListView } = await import('./pages/TransferTokens/ListView');

      return {
        Component: ProtectedListView,
      };
    },
    path: 'transfer-tokens',
  },
  {
    lazy: async () => {
      const { ProtectedEditView } = await import('./pages/TransferTokens/EditView');

      return {
        Component: ProtectedEditView,
      };
    },
    path: 'transfer-tokens/:id',
  },
  {
    lazy: async () => {
      const { ProtectedInstalledPlugins } = await import('./pages/InstalledPlugins');

      return {
        Component: ProtectedInstalledPlugins,
      };
    },
    path: 'list-plugins',
  },

  {
    lazy: async () => {
      const { PurchaseAuditLogs } = await import('./pages/PurchaseAuditLogs');

      return {
        Component: PurchaseAuditLogs,
      };
    },
    path: 'purchase-audit-logs',
  },
  {
    lazy: async () => {
      const { PurchaseSingleSignOn } = await import('./pages/PurchaseSingleSignOn');

      return {
        Component: PurchaseSingleSignOn,
      };
    },
    path: 'purchase-single-sign-on',
  },
  {
    lazy: async () => {
      const { PurchaseContentHistory } = await import('./pages/PurchaseContentHistory');

      return {
        Component: PurchaseContentHistory,
      };
    },
    path: 'purchase-content-history',
  },
];
