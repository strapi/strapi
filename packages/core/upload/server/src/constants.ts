const ACTIONS = {
  read: 'plugin::upload.read',
  readSettings: 'plugin::upload.settings.read',
  create: 'plugin::upload.assets.create',
  update: 'plugin::upload.assets.update',
  download: 'plugin::upload.assets.download',
  copyLink: 'plugin::upload.assets.copy-link',
  configureView: 'plugin::upload.configure-view',
};

const ALLOWED_SORT_STRINGS = [
  'createdAt:DESC',
  'createdAt:ASC',
  'name:ASC',
  'name:DESC',
  'updatedAt:DESC',
  'updatedAt:ASC',
];

const ALLOWED_WEBHOOK_EVENTS = {
  MEDIA_CREATE: 'media.create',
  MEDIA_UPDATE: 'media.update',
  MEDIA_DELETE: 'media.delete',
};

const FOLDER_MODEL_UID = 'plugin::upload.folder';
const FILE_MODEL_UID = 'plugin::upload.file';
const API_UPLOAD_FOLDER_BASE_NAME = 'API Uploads';

export {
  ACTIONS,
  FOLDER_MODEL_UID,
  FILE_MODEL_UID,
  API_UPLOAD_FOLDER_BASE_NAME,
  ALLOWED_SORT_STRINGS,
  ALLOWED_WEBHOOK_EVENTS,
};
