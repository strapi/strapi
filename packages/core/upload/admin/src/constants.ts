import { getTrad } from './utils';

export enum AssetType {
  Video = 'video',
  Image = 'image',
  Document = 'doc',
  Audio = 'audio',
}

export enum AssetSource {
  Url = 'url',
  Computer = 'computer',
}

export const PERMISSIONS = {
  // This permission regards the main component (App) and is used to tell
  // If the plugin link should be displayed in the menu
  // And also if the plugin is accessible. This use case is found when a user types the url of the
  // plugin directly in the browser
  main: [
    { action: 'plugin::upload.read', subject: null },
    {
      action: 'plugin::upload.assets.create',
      subject: null,
    },
    {
      action: 'plugin::upload.assets.update',
      subject: null,
    },
  ],
  copyLink: [
    {
      action: 'plugin::upload.assets.copy-link',
      subject: null,
    },
  ],
  create: [
    {
      action: 'plugin::upload.assets.create',
      subject: null,
    },
  ],
  download: [
    {
      action: 'plugin::upload.assets.download',
      subject: null,
    },
  ],
  read: [{ action: 'plugin::upload.read', subject: null }],
  configureView: [{ action: 'plugin::upload.configure-view', subject: null }],
  settings: [{ action: 'plugin::upload.settings.read', subject: null }],
  update: [{ action: 'plugin::upload.assets.update', subject: null, fields: null }],
};

export const tableHeaders = [
  {
    name: 'preview',
    key: 'preview',
    metadatas: {
      label: { id: getTrad('list.table.header.preview'), defaultMessage: 'preview' },
      isSortable: false,
    },
    type: 'image',
  },
  {
    name: 'name',
    key: 'name',
    metadatas: {
      label: { id: getTrad('list.table.header.name'), defaultMessage: 'name' },
      isSortable: true,
    },
    type: 'text',
  },
  {
    name: 'ext',
    key: 'extension',
    metadatas: {
      label: { id: getTrad('list.table.header.ext'), defaultMessage: 'extension' },
      isSortable: false,
    },
    type: 'ext',
  },
  {
    name: 'size',
    key: 'size',
    metadatas: {
      label: { id: getTrad('list.table.header.size'), defaultMessage: 'size' },
      isSortable: false,
    },
    type: 'size',
  },
  {
    name: 'createdAt',
    key: 'createdAt',
    metadatas: {
      label: { id: getTrad('list.table.header.createdAt'), defaultMessage: 'created' },
      isSortable: true,
    },
    type: 'date',
  },
  {
    name: 'updatedAt',
    key: 'updatedAt',
    metadatas: {
      label: { id: getTrad('list.table.header.updatedAt'), defaultMessage: 'last update' },
      isSortable: true,
    },
    type: 'date',
  },
];

export const sortOptions = [
  { key: 'sort.created_at_desc', value: 'createdAt:DESC' },
  { key: 'sort.created_at_asc', value: 'createdAt:ASC' },
  { key: 'sort.name_asc', value: 'name:ASC' },
  { key: 'sort.name_desc', value: 'name:DESC' },
  { key: 'sort.updated_at_desc', value: 'updatedAt:DESC' },
  { key: 'sort.updated_at_asc', value: 'updatedAt:ASC' },
];

export const pageSizes = [10, 20, 50, 100];

export const localStorageKeys = {
  modalView: `STRAPI_UPLOAD_MODAL_VIEW`,
  view: `STRAPI_UPLOAD_LIBRARY_VIEW`,
};

export const viewOptions = {
  GRID: 0,
  LIST: 1,
};
