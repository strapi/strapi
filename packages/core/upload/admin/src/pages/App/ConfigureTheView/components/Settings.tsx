import { Box, Grid, SingleSelectOption, SingleSelect, Field } from '@strapi/design-system';
import { useIntl } from 'react-intl';

import { pageSizes, sortOptions } from '../../../../constants';
import { getTrad } from '../../../../utils';

import type { Configuration } from '../../../../../../shared/contracts/configuration';

interface SettingsProps {
  sort: string;
  pageSize: string | number;
  onChange: ({
    target: { name, value },
  }: {
    target: { name: keyof Configuration; value: string | number };
  }) => void;
}

const Settings = ({ sort = '', pageSize = 10, onChange }: SettingsProps) => {
  const { formatMessage } = useIntl();

  return (
    <Box
      background="neutral0"
      hasRadius
      shadow="tableShadow"
      paddingTop={6}
      paddingBottom={6}
      paddingLeft={7}
      paddingRight={7}
    >
      <Grid.Root gap={4}>
        <Grid.Item s={12} col={6} direction="column" alignItems="stretch">
          <Field.Root
            hint={formatMessage({
              id: getTrad('config.entries.note'),
              defaultMessage: 'Number of assets displayed by default in the Media Library',
            })}
            name="pageSize"
          >
            <Field.Label>
              {formatMessage({
                id: getTrad('config.entries.title'),
                defaultMessage: 'Entries per page',
              })}
            </Field.Label>
            <SingleSelect
              onChange={(value) => onChange({ target: { name: 'pageSize', value } })}
              value={pageSize}
            >
              {pageSizes.map((pageSize) => (
                <SingleSelectOption key={pageSize} value={pageSize}>
                  {pageSize}
                </SingleSelectOption>
              ))}
            </SingleSelect>
            <Field.Hint />
          </Field.Root>
        </Grid.Item>
        <Grid.Item s={12} col={6} direction="column" alignItems="stretch">
          <Field.Root
            hint={formatMessage({
              id: getTrad('config.note'),
              defaultMessage: 'Note: You can override this value in the media library.',
            })}
            name="sort"
          >
            <Field.Label>
              {formatMessage({
                id: getTrad('config.sort.title'),
                defaultMessage: 'Default sort order',
              })}
            </Field.Label>
            <SingleSelect
              onChange={(value) => onChange({ target: { name: 'sort', value } })}
              value={sort}
              test-sort={sort}
              data-testid="sort-select"
            >
              {sortOptions.map((filter) => (
                <SingleSelectOption
                  data-testid={`sort-option-${filter.value}`}
                  key={filter.key}
                  value={filter.value}
                >
                  {formatMessage({ id: getTrad(filter.key), defaultMessage: `${filter.value}` })}
                </SingleSelectOption>
              ))}
            </SingleSelect>
            <Field.Hint />
          </Field.Root>
        </Grid.Item>
      </Grid.Root>
    </Box>
  );
};

export { Settings };
