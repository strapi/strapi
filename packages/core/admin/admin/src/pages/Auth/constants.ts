import type { ComponentType } from 'react';

import { ForgotPassword } from './components/ForgotPassword';
import { ForgotPasswordSuccess } from './components/ForgotPasswordSuccess';
import { Oops } from './components/Oops';
import { Register, RegisterProps } from './components/Register';
import { ResetPassword } from './components/ResetPassword';

export type AuthType =
  | 'login'
  | 'register'
  | 'register-admin'
  | 'forgot-password'
  | 'reset-password'
  | 'forgot-password-success'
  | 'oops'
  | 'providers';

export type FormDictionary = Record<AuthType, ComponentType | ComponentType<RegisterProps>>;

export const FORMS = {
  'forgot-password': ForgotPassword,
  'forgot-password-success': ForgotPasswordSuccess,
  // the `Component` attribute is set after all forms and CE/EE components are loaded, but since we
  // are here outside of a React component we can not use the hook directly
  login: () => null,
  oops: Oops,
  register: Register,
  'register-admin': Register,
  'reset-password': ResetPassword,
  providers: () => null,
} satisfies FormDictionary;
