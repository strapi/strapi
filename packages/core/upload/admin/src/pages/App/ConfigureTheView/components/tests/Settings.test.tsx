import { DesignSystemProvider } from '@strapi/design-system';
import { render as renderRTL } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router-dom';

import { pageSizes, sortOptions } from '../../../../../constants';
import { Settings } from '../Settings';

const testPageSize = pageSizes[0];
const testSort = sortOptions[0].value;

const render = ({ pageSize = testPageSize, sort = testSort, onChange = jest.fn() } = {}) => ({
  user: userEvent.setup(),
  ...renderRTL(<Settings sort={sort} pageSize={pageSize} onChange={onChange} />, {
    wrapper: ({ children }) => (
      <IntlProvider locale="en" messages={{}}>
        <DesignSystemProvider>
          <MemoryRouter>{children}</MemoryRouter>
        </DesignSystemProvider>
      </IntlProvider>
    ),
  }),
});

describe('Upload - Configure | Settings', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('initial render', () => {
    it('renders and matches the snapshot', async () => {
      const { getByText, container } = render();

      expect(getByText('Entries per page')).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });
  });

  describe('pageSize', () => {
    it('renders all page Sizes', async () => {
      const { getByRole, user } = render();

      await user.click(getByRole('combobox', { name: 'Entries per page' }));

      pageSizes.forEach((size) => {
        expect(getByRole('option', { name: size.toString() })).toBeInTheDocument();
      });
    });

    it('call onChange when changing page size', async () => {
      const onChange = jest.fn();
      const { user, getByRole } = render({ onChange });

      const testValue = pageSizes[1];

      await user.click(getByRole('combobox', { name: 'Entries per page' }));

      await user.click(getByRole('option', { name: testValue.toString() }));

      expect(onChange).toHaveBeenCalledTimes(1);
      expect(onChange).toHaveBeenCalledWith({ target: { name: 'pageSize', value: testValue } });
    });
  });

  describe('sort', () => {
    it('renders all sort options', async () => {
      const { user, getByRole } = render();

      await user.click(getByRole('combobox', { name: 'Default sort order' }));

      sortOptions.forEach((filter) => {
        expect(getByRole('option', { name: filter.value })).toBeInTheDocument();
      });
    });

    it('call onChange when changing sort', async () => {
      const onChange = jest.fn();
      const { user, getByRole } = render({ onChange });

      const testValue = sortOptions[1].value;

      await user.click(getByRole('combobox', { name: 'Default sort order' }));

      await user.click(getByRole('option', { name: testValue }));

      expect(onChange).toHaveBeenCalledTimes(1);
      expect(onChange).toHaveBeenCalledWith({ target: { name: 'sort', value: testValue } });
    });
  });
});
