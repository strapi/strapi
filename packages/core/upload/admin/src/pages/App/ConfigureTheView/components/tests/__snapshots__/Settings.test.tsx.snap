// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload - Configure | Settings initial render renders and matches the snapshot 1`] = `
.c0 {
  padding-block-start: 24px;
  padding-block-end: 24px;
  padding-inline-start: 32px;
  padding-inline-end: 32px;
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33, 33, 52, 0.1);
}

.c7 {
  background: #ffffff;
  border-radius: 4px;
  cursor: default;
  position: relative;
  overflow: hidden;
}

.c10 {
  flex: 1;
}

.c2 {
  align-items: stretch;
  flex-direction: column;
  display: flex;
}

.c4 {
  gap: 4px;
  align-items: stretch;
  flex-direction: column;
  display: flex;
}

.c8 {
  gap: 16px;
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
  display: flex;
}

.c11 {
  gap: 12px;
  align-items: center;
  flex-direction: row;
  display: flex;
}

.c5 {
  font-size: 1.2rem;
  line-height: 1.33;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
  font-weight: 600;
}

.c12 {
  font-size: 1.4rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
}

.c16 {
  font-size: 1.2rem;
  line-height: 1.33;
  color: #666687;
}

.c17 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c6 {
  display: flex;
}

.c9 {
  border: 1px solid #dcdce4;
  padding-block: 8px;
  padding-inline-start: 16px;
  padding-inline-end: 12px;
  cursor: pointer;
  outline: none;
  box-shadow: none;
  transition-property: border-color,box-shadow,fill;
  transition-duration: 0.2s;
}

.c9[aria-disabled='true'] {
  color: #8e8ea9;
}

.c9:focus-visible {
  outline: none;
}

.c9:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c15 {
  display: flex;
}

.c15>svg {
  fill: #8e8ea9;
}

.c13 {
  flex: 1;
  font-size: 1.4rem;
  line-height: 2.2rem;
  min-height: 2.2rem;
}

.c14 {
  display: flex;
  gap: 4px;
  flex-wrap: wrap;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  gap: 16px;
}

.c3 {
  grid-column: span 12;
  max-width: 100%;
}

@media (prefers-reduced-motion: no-preference) {

}

@media (min-width: 520px) {
  .c3 {
    grid-column: span 12;
  }
}

@media (min-width: 768px) {
  .c3 {
    grid-column: span 12;
  }
}

@media (min-width: 1080px) {
  .c3 {
    grid-column: span 6;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      >
        <div
          class="c4"
        >
          <label
            class="c5 c6"
            for=":r0:"
            id=":r0:-label"
          >
            Entries per page
          </label>
          <div
            aria-autocomplete="none"
            aria-controls="radix-:r2:"
            aria-describedby=":r0:-hint"
            aria-expanded="false"
            aria-labelledby=":r0:-label"
            aria-required="false"
            class="c7 c8 c9"
            data-state="closed"
            dir="ltr"
            name="pageSize"
            role="combobox"
            tabindex="0"
          >
            <span
              class="c10 c11"
            >
              <span
                class="c12 c13"
              >
                <span
                  class="c14"
                >
                  10
                </span>
              </span>
            </span>
            <span
              class="c11"
            >
              <span
                aria-hidden="true"
                class="c15"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                  />
                </svg>
              </span>
            </span>
          </div>
          <p
            class="c16"
            id=":r0:-hint"
          >
            Number of assets displayed by default in the Media Library
          </p>
        </div>
      </div>
      <div
        class="c2 c3"
      >
        <div
          class="c4"
        >
          <label
            class="c5 c6"
            for=":r3:"
            id=":r3:-label"
          >
            Default sort order
          </label>
          <div
            aria-autocomplete="none"
            aria-controls="radix-:r5:"
            aria-describedby=":r3:-hint"
            aria-expanded="false"
            aria-labelledby=":r3:-label"
            aria-required="false"
            class="c7 c8 c9"
            data-state="closed"
            dir="ltr"
            name="sort"
            role="combobox"
            tabindex="0"
          >
            <span
              class="c10 c11"
            >
              <span
                class="c12 c13"
              >
                <span
                  class="c14"
                >
                  createdAt:DESC
                </span>
              </span>
            </span>
            <span
              class="c11"
            >
              <span
                aria-hidden="true"
                class="c15"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                  />
                </svg>
              </span>
            </span>
          </div>
          <p
            class="c16"
            id=":r3:-hint"
          >
            Note: You can override this value in the media library.
          </p>
        </div>
      </div>
    </div>
  </div>
  <span
    class="c17"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
